core/src/ast/types.rs

```diff
@@ -641,6 +643,74 @@ impl TypeName {
         self.check_option(errors);
     }
 
+    /// Checks the validity of return types.
+    ///
+    /// This is equivalent to `TypeName::check_validity`, but it also ensures
+    /// that the type doesn't elide any lifetimes.
+    ///
+    /// Once we decide to support lifetime elision in return types, this function
```

Comment:
nit: worth filing a followup for this and linking to it

Reply:
Done: #160

Reply:
(in the code as well, you can also just mention the issue number)